packages/storage-plus/src/keys.rs

```diff
@@ -264,6 +265,42 @@ impl<T: Endian> From<IntKey<T>> for Vec<u8> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct TimestampKey(U64Key);
+
+impl TimestampKey {
+    pub fn new(ts: Timestamp) -> Self {
+        Self(ts.nanos().into())
+    }
+}
+
+impl<'a> PrimaryKey<'a> for TimestampKey {
+    type Prefix = ();
+    type SubPrefix = ();
+
+    fn key(&self) -> Vec<&[u8]> {
+        self.0.key()
+    }
+}
+
+impl<'a> Prefixer<'a> for TimestampKey {
+    fn prefix(&self) -> Vec<&[u8]> {
+        self.0.key()
+    }
+}
+
+impl From<Vec<u8>> for TimestampKey {
```

Comment:
This has the same issue as Intel from above.

I would make this a separate PR (these two functions), rest is fine to merge